src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/TemplateBuilder.cs

```diff
@@ -108,6 +96,37 @@ public IHtmlContent Build()
             // though _model may have been reset to null. Otherwise we might lose track of the model type /property.
             viewData.ModelExplorer = _modelExplorer.GetExplorerForModel(_model);
 
+            var formattedModelValue = viewData.Model;
+            var modelEnum = viewData.Model as Enum;
```

Comment:
Prefer this variable inside an `enum`-specific block and with a comment describing the case where it's needed e.g.
``` c#
if (_model == null)
{
    if (_readonly)
    {
        // ...
    }
}
else if (viewData.ModelMetadata.IsEnum)
{
    // Cover the case where something associated a string value with metadata for an enum type.
    var modelEnum = viewData.Model as Enum;
    if (modelEnum != null)
    {
        // ...
    }
}
```